tokio/src/io/driver/mod.rs

```diff
@@ -279,6 +290,25 @@ cfg_not_rt! {
     }
 }
 
+cfg_metrics! {
+   impl Handle {
+        // TODO: Remove this when handle contains `Arc<Inner>` so that we can return
+        // &IoDriverMetrics instead of using a closure.
+        //
+        // Related issue: https://github.com/tokio-rs/tokio/issues/4509
+        pub(crate) fn with_io_driver_metrics<F, R>(&self, f: F) -> Option<R>
```

Comment:
Fwiw, calling `self.inner()` does an arc bump anyway. I probably wouldn't bother w/ the closure, but it isn't a bit deal.

Reply:
Yeah, plan to refactor with removal of weak from io driver handle.